rpc/namenode.go

```diff
@@ -145,25 +145,32 @@ func (c *NamenodeConnection) resolveConnection() error {
 	}
 
 	var err error
+	stackErrors := fmt.Errorf("resolveConnection failed. ")
 
 	if c.host != nil {
 		err = c.host.lastError
 	}
 
-	for _, host := range c.hostList {
+	for i, host := range c.hostList {
 		if host.lastErrorAt.After(time.Now().Add(-backoffDuration)) {
+			stackErrors = fmt.Errorf("%sAttempt %v: still backing off host %v after error" +
+				" \"%s\" at time %s. ", stackErrors, i, host.address, host.lastError, host.lastErrorAt)
 			continue
 		}
 
 		c.host = host
 		c.conn, err = net.DialTimeout("tcp", host.address, connectTimeout)
 		if err != nil {
+			stackErrors = fmt.Errorf("%sAttempt %v: failed to dial host %v with error \"%s\". ",
+				stackErrors, i, host.address, err)
 			c.markFailure(err)
 			continue
 		}
 
 		err = c.writeNamenodeHandshake()
 		if err != nil {
+			stackErrors = fmt.Errorf("%sAttempt %v: failed namenode handshake to host %v with" +
```

Comment:
As discussed adding strings together like this could get problematic if they start growing in size

Reply:
The number of namenodes in a HA setup is pretty low and even in a federated setup we would only be dialling the namenode and standby in charge of the block pool we want to access.